WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/DesignPreviewFragment.kt

```diff
@@ -0,0 +1,115 @@
+ */
+class DesignPreviewFragment : FullscreenBottomSheetDialogFragment() {
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var viewModel: HomePagePickerViewModel
+
+    private lateinit var template: String
+    private lateinit var url: String
+
+    companion object {
+        const val DESIGN_PREVIEW_TAG = "DESIGN_PREVIEW_TAG"
+        private const val DESIGN_PREVIEW_TEMPLATE = "DESIGN_PREVIEW_TEMPLATE"
+        private const val DESIGN_PREVIEW_URL = "DESIGN_PREVIEW_URL"
+
+        fun newInstance(template: String, url: String) = DesignPreviewFragment().apply {
+            arguments = Bundle().apply {
+                putString(DESIGN_PREVIEW_TEMPLATE, template)
+                putString(DESIGN_PREVIEW_URL, url)
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            template = it.getString(DESIGN_PREVIEW_TEMPLATE, "")
+            url = it.getString(DESIGN_PREVIEW_URL, "")
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
+            inflater.inflate(R.layout.home_page_picker_preview_fragment, container)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel = ViewModelProviders.of(requireActivity(), viewModelFactory).get(HomePagePickerViewModel::class.java)
+
+        viewModel.previewState.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                is Loading -> {
+                    progressBar.setVisible(true)
+                    webView.setVisible(true)
+                    errorView.setVisible(false)
+                    webView.loadUrl(url)
+                }
+                is Loaded -> {
+                    progressBar.setVisible(false)
+                    webView.setVisible(true)
+                    errorView.setVisible(false)
+                }
+                is Error -> {
+                    progressBar.setVisible(false)
+                    webView.setVisible(false)
+                    errorView.setVisible(true)
+                    state.toast?.let { ToastUtils.showToast(requireContext(), it) }
+                }
+            }
+        }
+
+        backButton.setOnClickListener { closeModal() }
+
+        chooseButton.setOnClickListener { viewModel.onPreviewChooseTapped() }
+
+        webView.webViewClient = object : WebViewClient() {
+            override fun onPageFinished(view: WebView?, url: String?) {
+                super.onPageFinished(view, url)
+                viewModel.onPreviewLoaded(template)
+            }
+
+            override fun onReceivedError(view: WebView?, request: WebResourceRequest?, error: WebResourceError?) {
+                super.onReceivedError(view, request, error)
+                viewModel.onPreviewError()
```

Comment:
Hey @mkevins 👋 ,
thank you for reviewing this 🙇 

TBH, I had the same concern on this. I would prefer if [WebViewClient](https://developer.android.com/reference/android/webkit/WebViewClient) had a method denoting fatal failure only. I'm not sure if I can filter this out further using the error code from the [WebResourceError](https://developer.android.com/reference/android/webkit/WebResourceError) parameter 🤔

Reply:
Not sure if it makes sense to do some filtering based on the `request.url`.. in any case, I don't think it's a blocker here, but maybe just worth noting in case we see spurious errors being tracked. The only kind that come to mind are broken images.. but I guess those would not be spurious in this case (i.e. we'd want to know about that) since the preview is displaying content that is not generated by the user.